src/components/BoltGUI.tsx

```diff
@@ -0,0 +1,289 @@
+import { Label, TextInputComponent, ButtonInputComponent } from './inputs';
+import React, { Component } from 'react';
+
+import { MainPanel } from './MainPanel';
+import { TabData } from './ContentPanel';
+import goodboyLogo from '../assets/goodboy-logo.png';
+import { render } from 'react-dom';
+
+export interface BoltProps
+{
+    stage?: any; // this will be the reference object for the game if needed
+}
+
+export interface BoltState
+{
+    expandedX: boolean;
+    expandedY: boolean;
+    activeTab: number;
+}
+
+export class BoltGUI extends Component<BoltProps, BoltState>
+{
+    private tabs: TabData[] = [];
+
+    constructor(props: BoltProps)
+    {
+        super(props);
+        this.state = {
+            expandedX: true,
+            expandedY: true,
+            activeTab: 0,
+        };
+
+        const logInputChange = (value:any) => console.log(value);
+        
+        this.tabs.push({
+            title: 'Example',
```

Comment:
Just an idea, feel free to ignore if does not make sense: is it possible to put tab data into a separate file, like tab-data js/json? That could provide a good idea on what is the editable part.